src/main/java/org/junitpioneer/jupiter/converter/ByteArrayArgumentConverter.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.converter;
+
+import static java.lang.String.format;
+import static org.junitpioneer.jupiter.converter.ByteArrayConversion.ByteOrder.BIG_ENDIAN;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import org.junit.jupiter.params.converter.ArgumentConversionException;
+import org.junit.jupiter.params.converter.TypedArgumentConverter;
+import org.junit.jupiter.params.support.AnnotationConsumer;
+
+class ByteArrayArgumentConverter extends TypedArgumentConverter<Number, byte[]>
+		implements AnnotationConsumer<ByteArrayConversion> {
+
+	private ByteOrder order;
+
+	public ByteArrayArgumentConverter() {
+		super(Number.class, byte[].class);
+	}
+
+	@Override
+	public void accept(ByteArrayConversion annotation) {
+		this.order = getByteOrder(annotation);
+	}
+
+	@Override
+	protected byte[] convert(Number source) throws ArgumentConversionException {
+		if (source instanceof Byte) {
+			var bytes = (byte) source;
+			return ByteBuffer.allocate(Byte.BYTES).order(order).put(bytes).array();
+		} else if (source instanceof Short) {
+			var bytes = (short) source;
+			return ByteBuffer.allocate(Short.BYTES).order(order).putShort(bytes).array();
+		} else if (source instanceof Integer) {
+			var bytes = (int) source;
+			return ByteBuffer.allocate(Integer.BYTES).order(order).putInt(bytes).array();
+		} else if (source instanceof Long) {
+			var bytes = (long) source;
+			return ByteBuffer.allocate(Long.BYTES).order(order).putLong(bytes).array();
+		}
+		throw new ArgumentConversionException(format("Unsupported parameter type: %s", source.getClass()));
```

Comment:
Would you like to add support also for  `Float` and `Double`? Personally, I have no concrete use case for them but it would be trivial since `ByteBuffer` already supports them.

Reply:
If it's easy to implement we should support it right from the start  I guess.

Reply:
Actually, I was also thinking about wheter this converter should be generalized further, adding the support from `String` to `byte[]` via [`getBytes(String)`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#getBytes(java.lang.String)).

Right now, just from the `@ByteArrayConversion` name is not clear why the conversion is limited to numeric types (naming was my proposal so definitely my fault 😅).

I see two options:
* `@ByteArrayConversion` renamed to `@IntegralToByteArrayConversion` or `@NumberToByteArrayConversion` (in case float and double are added)
* Add support for string conversion with an additional `charset` attribute, which would be `""` by default (implying [`getBytes()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html#getBytes()) usage).

In case of the latter, `byteOrder` would be ignored when there is a string input, while `charset` would be ignored for numeric ones. Maybe not ideal but somehow similar to what [`@ValueSource`](https://junit.org/junit5/docs/5.9.3/api/org.junit.jupiter.params/org/junit/jupiter/params/provider/ValueSource.html) does today.

WDYT?

Reply:
Personally, I'd prefer not to include `getBytes` just to make sure that no one thinks that it's a number conversion, e.g.: `String` -> `BigDecimal` -> `byte[]`